src/github.com/matrix-org/dendrite/clientapi/writers/createroom.go

```diff
@@ -176,7 +177,19 @@ func createRoom(req *http.Request, cfg config.ClientAPI, roomID string) util.JSO
 		// Add the event to the list of auth events
 		builtEventMap[common.StateKeyTuple{e.Type, e.StateKey}] = ev
 		builtEvents = append(builtEvents, ev)
+	}
 
+	// send events to the room server
+	var m sarama.ProducerMessage
+	value, err := json.Marshal(builtEvents)
+	if err != nil {
+		return util.ErrorResponse(err)
+	}
+	m.Topic = "clientapiOutput" // TODO: Make this customisable like roomserver is?
+	m.Key = sarama.StringEncoder("")
+	m.Value = sarama.ByteEncoder(value)
+	if _, _, err = producer.SendMessage(&m); err != nil {
```

Comment:
You will need to send a message per event. `producer.SendMessages` provides a bulk send API.